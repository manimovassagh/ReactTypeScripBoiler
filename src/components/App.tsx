import React, {ReactElement} from 'react';
import StoreProvider from '../Store';

import Layout from './Layout';
import Routes from './Routes';

export default function App(): ReactElement {

  return (
    <StoreProvider>
      <Layout>
        <Routes />
      </Layout>
    </StoreProvider>
  )
}
