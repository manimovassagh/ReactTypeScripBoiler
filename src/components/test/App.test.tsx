import React from 'react';
import {render, RenderResult, matchSnapshot, act, mockAxios} from './test-utils';
import '@testing-library/jest-dom';

import App from '../App';

describe('<App /> without Routing', () => {
  let documentBody: RenderResult
  beforeEach(() => {
    documentBody = render(<App />);
  });

  test('matches snapshot', () => matchSnapshot(documentBody));

  test('renders: Willkommen beim BookMonkey from Home Component', () => {
    const textElement = documentBody.getByText(/Willkommen beim BookMonkey/i);
    expect(textElement).toBeInTheDocument();
  })
})

describe('<App /> with Routing', () => {
  const helper = async (route: string, text: string, needsAxios = false): Promise<void> => {
    if (needsAxios) mockAxios();
    let documentBody: RenderResult
    console.log('before act')
    act(async () => {documentBody = render(<App />, {route})}).then(() => {
      console.log('in act')
      if (documentBody) {
        const linkElement = documentBody.getByText(new RegExp(text, 'i'));
        expect(linkElement).toBeInTheDocument();
      }
    })
  }

  test('navigate to BookCreate', () => {
    const documentBody = render(<App />, {route: '/books/create'})
    const linkElement = documentBody.getByText(/Buch anlegen/i);
    expect(linkElement).toBeInTheDocument();
  })

  // @todo, fix UnhandledPromiseRejection
  // test('navigate to BookEdit', () => {
  //   helper('/books/edit/9783864906466', 'Buch editieren', true)
  // })

  // test('navigate to BookDetails', () => {
  //   helper('/books/9783864906466', 'Book Details', true)
  // })

  // test('navigate to BookList', () => {
  //   helper('/books', 'Book List', true)
  // })

  // test('navigate to Cart', () => {
  //   helper('/cart', 'Shopping Cart', true)
  // })

  // test('navigate to Home', () => {
  //   helper('/home', 'Willkommen beim BookMonkey')
  // })
})
