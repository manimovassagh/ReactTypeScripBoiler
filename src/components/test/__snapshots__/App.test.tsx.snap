// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> without Routing matches snapshot 1`] = `
<body>
  <div>
    <div
      class="ui menu"
    >
      <a
        aria-current="page"
        class="item active"
        href="/home"
      >
        Home
      </a>
      <a
        class="item"
        href="/books"
      >
        Books
      </a>
      <a
        class="item"
        href="/books/create"
      >
        new Book
      </a>
      <a
        class="item"
        href="/cart"
      >
        Shopping Cart
      </a>
      <div
        class="ui search item right"
      >
        <div
          class="ui icon input"
        >
          <input
            class="prompt"
            type="text"
            value=""
          />
          <i
            class="search icon"
          />
        </div>
      </div>
    </div>
    <div
      class="ui container"
    >
      <h4>
        Willkommen beim BookMonkey
      </h4>
      <a
        class="ui red button"
        href="/books"
      >
        Buchliste ansehen
      </a>
      <h2>
        Book Search
      </h2>
      <div
        class="ui search undefined"
      >
        <div
          class="ui icon input"
        >
          <input
            class="prompt"
            type="text"
            value=""
          />
          <i
            class="search icon"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
